### ISSUE TITLE
Custom pin icon looks large and blury on iOS.

### ISSUE THREAD

**Issue by kyurkchyan** [TorbenK/TK.CustomMap#100]
When you set an image to MKAnnotationView it's bounds are  set to the size of the image. Imagine you have an icon 60x120 in scale factor of 1. In scale factor of 2 the image size would be 120x240. When you load this image from the StreamImageSource for instance and set it to UIImage, MKAnnotationView will be sized accordingly - 120x240, however it should have the size 64x120, otherwise the image of pin will look large and blury. This can be fixed by taking into account the scale of the screen and setting the size of the MKAnnotationView to image.size divided by the scale factor. I replaced some code in iOS TKCustomMapRenderer with something like this and it worked perfect. 

`Device.BeginInvokeOnMainThread(() =>
                {
                    annotationView.Image = image;
                    var scale = UIScreen.MainScreen.Scale;
                    annotationView.Bounds = new CGRect(0, 0,
                        image.Size.Width / scale,
                        image.Size.Height / scale);
                });`

Take a look to the screenshots below which show the difference of how the images are rendered.
![pin](https://cloud.githubusercontent.com/assets/3243139/16559456/1608c1ca-41fe-11e6-9869-910fc44d1a48.png)

**Comment by TorbenK** [comment]
I would have guessed this happens automatically via `@2x` file extension?

**Comment by kyurkchyan** [comment]
Yeah, if you load from file perhaps. But if you use StreamImageSource (which I did), there is no a "filename", from which UIImage could guess it's scale, there's just stream of bytes. Perhaps this issue pertains just to StreamImageSource.

**Comment by TorbenK** [comment]
Ok so what we need is a `AutoScale` property I guess. Thank you!

**Comment by kyurkchyan** [comment]
Yep, exactly :) thank you too

**Comment by EdHubbell** [comment]
I'm having this same problem. Is there an easier solution than forking? I hate to put myself in a position where I fall behind on new releases and fixes. I thought I'd try to override UpdateImage in a new custom renderer, but I can't seem to get that to go.  

Great control, BTW - Quite useful. Thanks for the hard work. ~Ed

On second thought, please disregard. I've got to cache images anyway, so I'll try to move to a file based ImageSource and cache @2x and @3x versions of the custom pin images. But if there is a solution, feel free to elaborate.
